Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -162,14 +162,25 @@ public AndConstraint<TAssertions> NotContainNulls(string because = "", params ob
                     .FailWith("Expected {context:collection} not to contain nulls{reason}, but collection is <null>.");
             }
 
-            object[] values = Subject.Cast<object>().ToArray();
-            for (int index = 0; index < values.Length; index++)
+            IEnumerable<int> indices = Subject
+                .Cast<object>()
+                .Select((item, index) => new { Item = item, Index = index })
+                .Where(e => ReferenceEquals(e.Item, null))
+                .Select(e => e.Index);
+
+            if (indices.Any())
             {
-                if (ReferenceEquals(values[index], null))
+                if (indices.Count() > 1)
```

Comment:
We had some performance reports for some specific methods. However, this specific one will not suffer from that. But a simple `ToArray` would be fine.